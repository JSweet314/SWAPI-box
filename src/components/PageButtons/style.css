.page-buttons {
  width: 100%;
  display: flex;
  justify-content: space-evenly;
  align-items: center;
}

.page-buttons--beginning {
  justify-content: center;
}

.page-buttons button {
  font-size: 2.5rem;
  margin: 2rem;
}