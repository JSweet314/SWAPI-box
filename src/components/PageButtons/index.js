import React from 'react';
import PropTypes from 'prop-types';
import './style.css';

const PageButtons = ({pageNumber}) => {
  if (pageNumber === 1) {
    return (
      <div className="page-buttons page-buttons--beginning">
        <button>Next</button>
      </div>
    );
  }

  return (
    <div className="page-buttons">
      <button>Back</button>
      <button>Next</button>
    </div>
  );
};

PageButtons.propTypes = {
  pageNumber: PropTypes.number.isRequired
};

export default PageButtons;