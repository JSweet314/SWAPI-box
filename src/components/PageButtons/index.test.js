import React from 'react';
import {shallow} from 'enzyme';
import PageButtons from './index';

describe('PageButtons', () => {
  /*eslint-disable no-undef*/
  const mockPageButtonClick = jest.fn();
  /*eslint-enable no-undef*/

  it('should match a snapshot with no previous url passed', () => {
    const wrapper = shallow(
      <PageButtons
        handlePageButtonClick={mockPageButtonClick}
        next="www.whosthebestpuppy.com"
        previous={null} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should match a snapshot with no next url passed', () => {
    const wrapper = shallow(
      <PageButtons
        handlePageButtonClick={mockPageButtonClick}
        next={null}
        previous="www.spacejam.com" />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should match a snapshot with both prev. and next urls passed', () => {
    const wrapper = shallow(
      <PageButtons
        handlePageButtonClick={mockPageButtonClick}
        next="www.whosthebestpuppy.com"
        previous="www.spacejam.com" />
    );
    expect(wrapper).toMatchSnapshot();
  });
});