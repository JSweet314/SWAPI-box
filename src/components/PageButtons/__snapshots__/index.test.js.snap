// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageButtons should match a snapshot with both prev. and next urls passed 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PageButtons
    handlePageButtonClick={[Function]}
    next="www.whosthebestpuppy.com"
    previous="www.spacejam.com"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          name="previous"
          onClick={[Function]}
>
          Back
</button>,
        <button
          name="next"
          onClick={[Function]}
>
          Next
</button>,
      ],
      "className": "page-buttons",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Back",
          "name": "previous",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Back",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Next",
          "name": "next",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Next",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            name="previous"
            onClick={[Function]}
>
            Back
</button>,
          <button
            name="next"
            onClick={[Function]}
>
            Next
</button>,
        ],
        "className": "page-buttons",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Back",
            "name": "previous",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Back",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Next",
            "name": "next",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Next",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`PageButtons should match a snapshot with no next url passed 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PageButtons
    handlePageButtonClick={[Function]}
    next={null}
    previous="www.spacejam.com"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <button
        name="previous"
        onClick={[Function]}
>
        Back
</button>,
      "className": "page-buttons page-buttons--beginning",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Back",
        "name": "previous",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": "Back",
      "type": "button",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <button
          name="previous"
          onClick={[Function]}
>
          Back
</button>,
        "className": "page-buttons page-buttons--beginning",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Back",
          "name": "previous",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Back",
        "type": "button",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`PageButtons should match a snapshot with no previous url passed 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PageButtons
    handlePageButtonClick={[Function]}
    next="www.whosthebestpuppy.com"
    previous={null}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <button
        name="next"
        onClick={[Function]}
>
        Next
</button>,
      "className": "page-buttons page-buttons--beginning",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Next",
        "name": "next",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": "Next",
      "type": "button",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <button
          name="next"
          onClick={[Function]}
>
          Next
</button>,
        "className": "page-buttons page-buttons--beginning",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Next",
          "name": "next",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Next",
        "type": "button",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
